pages/s390.adoc

```diff
@@ -0,0 +1,133 @@
+Track:: A track contains various records, the size of a track is fixed and the last record
+is used to fill the remainder data (in the case where the track isn't divisible by whole
+records).
+Cylinder:: A cylinder can hold various tracks, this is the biggest unit for a CKD disk
+and has a fixed amount of tracks.
+
+== FBA Disk
+FBA (Fixed-Block-Addressing) disks operate on a similar fashion to their PC counterparts.
+
+== IPL
+The general procedure for running the mainframe is to IPL (Initial-Program-Loader) from a
+telephone-dial circle. Selecting the device number corresponding to the disk, typically
+its 1B9 on a lot of places. After that the operator will press the "IPL" button on the machine.
+
+Real mainframes can take from 30 seconds to 30 minutes to start, as they're not meant to
+shutdown/restart constantly.
+
+Under the hood, the mainframe will load the first cylinders of a disk onto the storage. For S/390
+and previous models the amount of loaded data from the disk is 4096 bytes, for z/Arch its 8192.
+
+From that point onwards, its the programmer's responsability to perform what it wishes to do,
+the IPL is always going to be smaller than a single record.
+
+== HLASM
+
++
+* The S/3X0 can be catalogued as a CISC architecture, it offers a wide repertoire of instructions
+each with a specific purpouse, mostly related to transactional operations.
+
+S/3X0 has various assemblers:
+z390:: An assembler written in Java.
+GNU AS:: Binutils implementation of S/3X0 assembly, not compatible with normal HLASM
+except on i370 targets.
+IF0X0:: Vanilla assembler provided by TK-4 and MVS3.8j.
+
+== Programming paradigms
+
+S/3X0 was commonly used along with procedular languages regularly, however they
+all shared common terminology, such as:
+
+Character:: 8-bits storage area.
+Halfword:: 16-bits storage area.
+Word:: 32-bits storage area.
+Double word:: 64-bits storage area.
+Define Storage (DS on HLASM):: Uninitialized data definition.
+Define Character (DC on HLASM):: Initialized data definition.
+Procedure:: A function.
+
+== Input/Output
+The Channel Subystem I/O is used instead of the PIO (Polling I/O) or MMIO (Memory-Mapped
+I/O); it operates based around batch processing and non-blocking I/O while keeping the
+transactional and reliability aspect.
+
+== Example OSes
+Many public domain Operating Systems exists for the mainframe, due to their publicly accessible
+nature, they're used as a reference and potential learning resources:
+
+MVS:: Multiple Virtual Storage. An Operating System that could isolate programs into different
+address spaces.
+DOS/VS:: Direct Operating System with Virtual Storage. An small operating system resembling
+MVS in functionality.
+TK-4:: A distrobution of MVS3.8j, includes the SPF editor (with support for lowercase characters
+and punch-card reordering).
+MTS:: Multi-Task System, self descriptive, one of the first instances of a multitasking OS in
+computer history.
```

Comment:
This definitely needs a source, I couldn't find any mainframe OS with this name.
I could only find a S/360-67 and S/370 OS from 1967 called "Michigan Terminal System (MTS
)"